### PR TITLE
initial code push

### PR DESCRIPTION
**Pull request by vc-vishva**
- perform crud task
- implement session and cookies
- integrete this api using html

### CODE REVIEW

File: sign-up task/.env
Comment (vc-mohit) [inline sign-up task/.env:1]: env file should not be pushed to git, put it in git ignore.

File: sign-up task/package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "bcrypt": "^5.1.0",
+    "cookie-parser": "^1.4.6",
+    "dotenv": "^16.0.3",
+    "express": "^4.18.2",
+    "express-session": "^1.17.3",
+    "session": "^0.1.0",
```
Comment (vc-mohit) [inline sign-up task/package.json:17]: this package is required?

File: sign-up task/dashboard.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  <script>
+    fetch('/user')
+      .then(response => response.json())
+      .then(data => {
+        document.getElementById('name').innerText = data.name;
+        document.getElementById('email').innerText = data.email;
+        document.getElementById('phone').innerText = data.phone;
+      })
+      .catch(error => {
+        console.log('Error:', error);
```
Comment (vc-mohit) [inline sign-up task/dashboard.html:27]: remove console

File: sign-up task/main.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+const fs = require("fs");
+const bcrypt = require("bcrypt");
+const cookieParser = require("cookie-parser");
+const app = express();
+const port = 3000;
+app.use(express.urlencoded({ extended: true }));
+app.use(cookieParser());
+app.use(
+  session({
+    secret: "myscreatekey",
```
Comment (vc-mohit) [inline sign-up task/main.js:14]: check spelling

File: sign-up task/main.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    saveUninitialized: false,
+    store: new FileStore({
+      path: path.join(path.resolve(), "../store-session/sessions"),
+    }),
+  })
+);
+//check login or not
+const checkLoggedIn = (req, res, next) => {
+  console.log(req.session.user, ",,,,,,,,,,,,,,,,,req.session.user");
+  if (req.session.user) {
```
Comment (parikshit-VC) [inline sign-up task/main.js:25]: check cookies too whether we have seesionId stored or not

File: sign-up task/main.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    next();
+  } else {
+    res.redirect("/login");
+  }
+};
+const checkNotLoggedIn = (req, res, next) => {
+  console.log(req.session.user, "req.session.user");
+  if (req.session.user) {
```
Comment (parikshit-VC) [inline sign-up task/main.js:33]: same check if cookies exist

**Comment by vc-mohit** [comment]
I am rejecting the PR because of the following reasons:
1. remove all consoles
2. why did you create two functions checkLoggedIn & checkNotLoggedIn
3. always use async methods
4. Code should be formatted and also readable

**Pull request closed without merging**